google/cloud/pubsub/internal/batching_publisher_tracing_connection_test.cc

```diff
@@ -0,0 +1,120 @@
+using ::google::cloud::testing_util::StatusIs;
+using ::google::cloud::testing_util::ThereIsAnActiveSpan;
+
+TEST(BatchingPublisherTracingConnectionTest, PublishSpan) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<MockPublisherConnection>();
+  EXPECT_CALL(*mock, Publish)
+      .WillOnce([&](pubsub::PublisherConnection::PublishParams const&) {
+        return google::cloud::make_ready_future(
+            google::cloud::StatusOr<std::string>("test-id-0"));
+      });
+  auto connection = MakeBatchingPublisherTracingConnection(std::move(mock));
+
+  auto response = connection
+                      ->Publish({pubsub::MessageBuilder{}
+                                     .SetData("test-data-0")
+                                     .SetOrderingKey("ordering-key-0")
+                                     .Build()})
+                      .get();
+
+  EXPECT_THAT(response, StatusIs(StatusCode::kOk));
+  EXPECT_THAT(
+      span_catcher->GetSpans(),
+      ElementsAre(AllOf(
+          SpanHasInstrumentationScope(), SpanKindIsClient(),
+          SpanNamed("pubsub::BatchingPublisherConnection::Publish"),
+          SpanWithStatus(opentelemetry::trace::StatusCode::kOk),
+          SpanHasAttributes(OTelAttribute<int>("gcloud.status_code", 0)))));
+}
+
+TEST(BatchingPublisherTracingConnectionTest, FlushSpan) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<MockPublisherConnection>();
+  EXPECT_CALL(*mock, Flush).Times(1);
+  auto connection = MakeBatchingPublisherTracingConnection(std::move(mock));
+
+  connection->Flush(PublisherConnection::FlushParams{});
+
+  EXPECT_THAT(
+      span_catcher->GetSpans(),
+      ElementsAre(AllOf(
+          SpanHasInstrumentationScope(), SpanKindIsClient(),
+          SpanNamed("pubsub::BatchingPublisherConnection::Flush"),
+          SpanWithStatus(opentelemetry::trace::StatusCode::kOk),
+          SpanHasAttributes(OTelAttribute<int>("gcloud.status_code", 0)))));
+}
+
+TEST(FlowControlledPublisher, ResumePublishSpan) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<MockPublisherConnection>();
+  EXPECT_CALL(*mock, ResumePublish).Times(1);
+  auto connection = MakeBatchingPublisherTracingConnection(std::move(mock));
+
+  connection->ResumePublish(PublisherConnection::ResumePublishParams{});
+
+  EXPECT_THAT(
+      span_catcher->GetSpans(),
+      ElementsAre(AllOf(
+          SpanHasInstrumentationScope(), SpanKindIsClient(),
+          SpanNamed("pubsub::BatchingPublisherConnection::ResumePublish"),
+          SpanWithStatus(opentelemetry::trace::StatusCode::kOk),
+          SpanHasAttributes(OTelAttribute<int>("gcloud.status_code", 0)))));
+}
+
+TEST(MakeBatchingPublisherTracingConnectionTest, CreateTracingConnection) {
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<MockPublisherConnection>();
+  EXPECT_CALL(*mock, Flush).WillOnce([] {
+    EXPECT_TRUE(ThereIsAnActiveSpan());
```

Comment:
This seems.... surprising? concerning?

We never set the span as active (e.g. by wrapping it in an `opentelemetry::trace::Scope`. I would expect this thing to be false.